tests/test_dragon_installer.py

```diff
@@ -0,0 +1,519 @@
+
+    utility = check_for_utility(ld_config)
+
+    assert not utility
+
+
+def test_check_for_utility_exists(test_dir: str) -> None:
+    """Ensure that looking for an existing utility returns a non-empty path"""
+    utility = check_for_utility("ls")
+
+    assert utility
+
+
+def test_is_crayex_missing_ldconfig(monkeypatch: pytest.MonkeyPatch) -> None:
+    """Ensure the cray ex platform check doesn't fail when ldconfig isn't
+    available for use"""
+
+    def mock_util_check(util: str) -> bool:
+        if util == "ldconfig":
+            return False
+        return True
+
+    with monkeypatch.context() as ctx:
+        # mock utility existence
+        ctx.setattr(
+            smartsim._core.entrypoints.dragon_install,
+            "check_for_utility",
+            mock_util_check,
+        )
+
+        is_cray = is_crayex_platform()
+        assert not is_cray
+
+
+def test_is_crayex_missing_fi_info(monkeypatch: pytest.MonkeyPatch) -> None:
+    """Ensure the cray ex platform check doesn't fail when fi_info isn't
+    available for use"""
+
+    def mock_util_check(util: str) -> bool:
+        if util == "fi_info":
+            return False
+        return True
+
+    with monkeypatch.context() as ctx:
+        # mock utility existence
+        ctx.setattr(
+            smartsim._core.entrypoints.dragon_install,
+            "check_for_utility",
+            mock_util_check,
+        )
+
+        is_cray = is_crayex_platform()
+        assert not is_cray
+
+
+@pytest.mark.parametrize(
+    "is_cray,output,return_code",
+    [
+        pytest.param(True, "cray pmi2.so\ncxi\ncray pmi.so\npni.so", 0, id="CrayEX"),
+        pytest.param(False, "cray pmi2.so\ncxi\npni.so", 0, id="No PMI"),
+        pytest.param(False, "cxi\ncray pmi.so\npni.so", 0, id="No PMI 2"),
+        pytest.param(False, "cray pmi2.so\ncray pmi.so\npni.so", 0, id="No CXI"),
+        pytest.param(False, "pmi.so\ncray pmi2.so\ncxi", 0, id="Non Cray PMI"),
+        pytest.param(False, "cray pmi.so\npmi2.so\ncxi", 0, id="Non Cray PMI2"),
+    ],
+)
+def test_is_cray_ex(
+    monkeypatch: pytest.MonkeyPatch, is_cray: bool, output: str, return_code: int
+) -> None:
+    """Test that cray ex platform check result is returned as expected"""
+
+    def mock_util_check(util: str) -> bool:
+        # mock that we have the necessary tools
+        return True
+
+    with monkeypatch.context() as ctx:
+        # make it look like the utilies always exist
+        ctx.setattr(
+            smartsim._core.entrypoints.dragon_install,
+            "check_for_utility",
+            mock_util_check,
+        )
+        # mock
+        ctx.setattr(
+            smartsim._core.entrypoints.dragon_install,
+            "_execute_platform_cmd",
+            lambda x: (output, return_code),
+        )
+
+        platform_result = is_crayex_platform()
+        assert is_cray == platform_result
+
+
+def test_install_package__no_wheel(test_dir: str):
```

Comment:
pedantic typo thing:
```suggestion
def test_install_package_no_wheel(test_dir: str):
```